src/market/api/api.ts

```diff
@@ -52,6 +52,8 @@ const matchSearch =
   };
 
 function distributedCopy(items: number[], n: number): number[] {
+  if (!items) return [];
+  if (items.length <= n) return items;
```

Comment:
This safety looks good for the `distributedCopy` function :+1: but does it help for the `sparklineAsSvgData` function below?

Reply:
its always called with data provided by this function so in this case yes :sweat_smile:

Reply:
line 185
```js

 sparklineIn7d: currency?.sparkline_in_7d?.price
        ? sparklineAsSvgData(
            distributedCopy(currency.sparkline_in_7d.price, 6 * 7)
          ) // keep 6 points per day
        : null,
```